File: src/main/java/com/oppshan/files/file/FileNodeView.java
package com.oppshan.files.file;

import io.quarkus.runtime.annotations.RegisterForReflection;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

import java.io.Serial;
import java.io.Serializable;
import java.time.Instant;
import java.util.UUID;

@Valid
@RegisterForReflection
public record FileNodeView(
        @NotNull
        UUID uuid,

        @NotEmpty
        String name,

        @NotEmpty
        String mimeType,

        boolean directory,

        @PositiveOrZero
        long sizeBytes,

        UUID parentUuid,

        @NotNull
        Instant createdAt,

        @NotNull
        Instant lastModifiedAt
) implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;
}

File: src/main/angular/src/app/models/file-node-view.ts
import {DateTime} from 'luxon';
import {TransformDateTime} from '../misc/transform-date-time';

export class FileNodeView {

  uuid!: string;

  name!: string;

  mimeType!: string;

  directory!: boolean;

  sizeBytes!: number;

  parentUuid!: string | null;

  @TransformDateTime()
  createdAt!: DateTime;

  @TransformDateTime()
  lastModifiedAt!: DateTime;
}
